File: src/gestorAplicacion/clasesPrincipales/Bus.java
package gestorAplicacion.clasesPrincipales;
import java.time.LocalDate;
import java.util.ArrayList;

public abstract class Bus {
	private Ruta rutaBus;
	protected int km;
	protected int precio;
	protected int capacidad;
	private int codigo; 
	private static ArrayList<String> sillaNoDisponibles=new ArrayList<String>();
	protected String modelo;
	protected String placa;
	protected Conductor conductor;
	private static  ArrayList<Bus> buses = new ArrayList<>(); //Agregue este atributo valen
	private ArrayList<String> sillas;
	//Constructor
	public Bus(String modelo, String placa, Conductor conductor, ArrayList<String> sillas){
		this.modelo=modelo;
		this.placa=placa;
		this.conductor=conductor;
		this.sillas=sillas;
		Bus.buses.add(this);
	}
	public Bus(String modelo, String placa, Conductor conductor, ArrayList<String> sillas,int capacidad){
		this.modelo=modelo;
		this.placa=placa;
		this.conductor=conductor;
		this.sillas=sillas;
		this.capacidad=capacidad;
		Bus.buses.add(this);
	}
	//get&set buses
	public static ArrayList<Bus> getBuses() {
		return buses;
	}
	public static void setBuses(ArrayList<Bus> buses) {
		Bus.buses = buses;
	}
	//get&set modelo
	public String getModelo() {
		return modelo;
	}
	public void setModelo(String modelo) {
		this.modelo=modelo;
	}
	//get&set placa
	public String getPlaca() {
		return modelo;
	}
	public void setPlaca(String placa) {
		this.placa=placa;
	}
	//get&set conductor
	public Conductor getConductor() {
		return conductor;
	}
	public void setConductor(Conductor conductor) {
		this.conductor=conductor;
	}
	
	//get&set sillas
	public ArrayList<String> getSillas() { //Agregue st y get de sillas y de buses
		return sillas;
	}
	public void setSillas(ArrayList<String> sillas) {
		this.sillas = sillas;
	}
	//get&set capacidad
	public int getCapacidad() {
		return capacidad;
	}
	public void setCapacidad(int capacidad) {
		this.capacidad=capacidad;
	}
	//get&set sillanodisponible
	public static ArrayList<String> getSillaNoDisponibles() {
		return sillaNoDisponibles;
	}

	public static void setSillaNoDisponibles(String sillaNoDisponibles) {
		Bus.sillaNoDisponibles.add(sillaNoDisponibles);
	}
	//get&set km
	public int getKm() {
		return km;
	}

	public void setKm(int km) {
		this.km = km;
	}
	//get&set precio
	public int getPrecio() {
		return precio;
	}

	public void setPrecio(int precio) {
		this.precio = precio;
	}
	
	
	
	public int getCodigo() {
		return codigo;
	}
	public void setCodigo(int codigo) {
		this.codigo = codigo;
	}
	public static void setSillaNoDisponibles(ArrayList<String> sillaNoDisponibles) {
		Bus.sillaNoDisponibles = sillaNoDisponibles;
	}
	//metodos abstractos
	protected abstract void sillasDisponibles();//params Viaje viaje
	protected abstract String sillaTiquete(Tiquete tiquete);
	
	
}

File: Python/Bus.py
class Bus:

    #Constructor
    def __init__(self,placa,conductor,sillas,km=0,precioKm=0,capacidad=0,codigo=0):
        self.km = 0
        self.precioKm = 0
        self.capacidad = 0
        self._codigo = 0
        self.placa = placa
        self.conductor = conductor
        self._sillas = sillas
        _buses = [] #Agregue este atributo valen
        _sillaNoDisponibles = []


    #get&set buses
    @staticmethod
    def getBuses():
        return Bus._buses
    @staticmethod
    def setBuses(buses):
        Bus._buses = buses

    #get&set placa
    def getPlaca(self):
        return self.placa
    def setPlaca(self, placa):
        self.placa=placa

    #get&set conductor
    def getConductor(self):
        return self.conductor
    def setConductor(self, conductor):
        self.conductor=conductor

    #get&set sillas
    def getSillas(self):
        return self._sillas
    def setSillas(self, sillas):
        self._sillas = sillas

    #get&set capacidad
    def getCapacidad(self):
        return self.capacidad
    def setCapacidad(self, capacidad):
        self.capacidad=capacidad

    #get&set sillanodisponible
    @staticmethod
    def getSillaNoDisponibles():
        return Bus._sillaNoDisponibles
    @staticmethod
    def setSillaNoDisponibles(sillaNoDisponibles):
        Bus._sillaNoDisponibles.append(sillaNoDisponibles)

    #get&set km
    def getKm(self):
        return self.km
    def setKm(self, km):
        self.km = km

    #get&set precio
    def getPrecioKm(self):
        return self.precioKm
    def setPrecioKm(self, precioKm):
        self.precioKm = precioKm

    #get&set precio
    def getCodigo(self):
        return self._codigo
    def setCodigo(self, codigo):
        self._codigo = codigo

